### DISCUSSION TITLE
Fails on HACS components

### DISCUSSION THREAD

**Discussion by luebbe** [frenck/action-home-assistant#20] [General]
As the subject says, the check fails on some custom components installed via HACS.

See: https://github.com/luebbe/HomeAssistantConfig/runs/1903752083?check_suite_focus=true

Is there any way to tell the check to ignore these and continue with the rest?
I understand that it is not easy/possible to handle non-core components in the check, but it would be nice if the check would detect this case and just throw a message along the lines "custom component detected: xyz" instead of failing alltogether.

Is this possible?

**Comment by frenck** [comment]
Unfortunately, that is not possible. The only solution is to commit the custom integrations into the repository.

Home Assistant tries to resolve integration that are not there. From its perspective, there is no difference in core or custom integrations, nor can it tell if it is wrong/broken or missing.

**Reply by luebbe** [reply]
Thank you, I kind of expected this. No Problem!

**Reply by stin7** [reply]
> The only solution is to commit the custom integrations into the repository.

Couple follow up questions if we want to silence these warnings by committing the custom integration:
1. Do we need to commit the relevant files in `custom_components` or another is there another location in the config directory where the integrations are stored?
2. If it is in the `custom_components` directory, does the whole custom integration need to be committed? Or just a subset of the files in `custom_components`? (e.g. just manifest.json)

Thanks
